client/my-sites/site-settings/settings-traffic/main.jsx

```diff
@@ -89,18 +89,10 @@ const SiteSettingsTraffic = ( {
 	);
 };
 
-const connectComponent = connect( state => {
-	const site = getSelectedSite( state );
-	const siteId = getSelectedSiteId( state );
-	const isJetpack = isJetpackSite( state, siteId );
-	const jetpackSettingsUiSupported = isJetpack && siteSupportsJetpackSettingsUi( state, siteId );
-
-	return {
-		site,
-		isJetpack,
-		jetpackSettingsUiSupported,
-	};
-} );
+const connectComponent = connect( state => ( {
+	isJetpack: isJetpackSite( state, getSelectedSiteId( state ) ),
+	site: getSelectedSite( state ),
```

Comment:
It should be possible to remove this `site` prop completely. It's used to:
- be passed to `SiteSettingsNavigation` as a prop, but that component is connected to Redux and gets `site` from there
- do something conditionally if `site` is truthy or falsy. But thanks to the `siteSelection` controller, `site` is guaranteed to exist and be valid (at least since #28449).

These facts can be used to eliminate a lot of code from settings.

Reply:
Good call! Removed all those in https://github.com/Automattic/wp-calypso/pull/30041/commits/f0bfbc826e71e4695862113d1044b49d8d23dcc8